### ISSUE TITLE
provide support for meta thing to the actor.

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#17487]
Current we could not extends ```MessageDispatcher``` .
what I want is provide some context thing to the an actor and all of its child ,but seems there is no convenient way to implement this.

1,I could not extends the ```ActorContext```,so I could not provide some additional context when the actor is created.
2,```ActorContext``` don't have any meta thing where we could put something when we create the actor.
3,```MessageDispatcher``` can't be extends,so I could not provide a wrapper of the runnable,which set up the ```ThreadLocal``` things and then clean up it or I may using the   ```protected[akka] def createMailbox(actor: Cell, mailboxType: MailboxType): Mailbox```  which created a dedicated special mailbox.

4,```mailbox``` maybe extends,but don't play nice here to.I could extends the mailbox, and then override ```run``` method,but accept just as an ```FQCN``` but not any provider ,and I could not provide any meta to it too.

In short:

I may extends ```Actor``` class and then return an ```ActorContext``` wrapper.
I may setup the additional context via an extends ```MessageDispatcher``` which provide an ```runnable``` wrapper,which capture the context things.or I could just override ```createMailbox``` method to return an special ```Mailbox```.
I may extends the ```Mailbox```,and setup the ThreadLocal things.

I think how about provide a meta support for the props/or the actor context,which an actor and all it's children inheritable and mergeable,like things we have in the stream,attributes?cause akka may deploy the actor to remote,I think the metathing we may should make sure the meta things are serialable. I don't want to passing the context-like reference around ,which the whole branch of the actor shared.

**Comment by hepin1989** [comment]
maybe I could do it via ```Deploy``` and support meta things in the ```config```.and when I create the child I create the props from the parent's props.

**Comment by hepin1989** [comment]
I final come up with that I could make use of the ```IndirectActorProducer```.
keep open here as it may nice to provide ```attributes``` for actors .

**Comment by rkuhn** [comment]
It is not yet entirely clear to me: why not put all configuration into the constructor argument list explicitly? Why hide meta-information and provide it via side-channels? This has never been a good idea AFAIK, it has only ever been used to work around legacy code restrictions.
